addons/io/org.openhab.io.hueemulation/src/main/java/org/openhab/io/hueemulation/internal/HueEmulationServlet.java

```diff
@@ -0,0 +1,517 @@
+     * @return
+     *         Map <item name, HueDevice>
+     */
+    private Map<String, HueDevice> getHueDevices() {
+        Collection<Item> items = getTaggedItems();
+        Map<String, HueDevice> devices = new HashMap<String, HueDevice>();
+        Iterator<Item> it = items.iterator();
+        while (it.hasNext()) {
+            Item item = it.next();
+            devices.put(item.getName(), itemToDevice(item));
+        }
+        return devices;
+    }
+
+    /**
+     * Returns the item name and voice name of each item
+     *
+     * @param username
+     * @return
+     *         Map<item name, item voice tag>
+     */
+    public Map<String, String> getHueDeviceNames() {
+        Collection<Item> items = getTaggedItems();
+        Map<String, String> devices = new HashMap<String, String>();
+        Iterator<Item> it = items.iterator();
+        while (it.hasNext()) {
+            Item item = it.next();
+            devices.put(item.getName(), item.getLabel());
+        }
+        return devices;
+    }
+
+    /**
+     * Converts an VoiceItem to a HueDevice
+     *
+     * @param voiceItem
+     * @return
+     *         HueDevice
+     */
+    private HueDevice itemToDevice(Item item) {
+        State itemState = item.getState();
+        short bri = 0;
+        if (itemState instanceof DecimalType) {
+            bri = (short) ((((DecimalType) itemState).intValue() * 255) / 100);
+        } else if (itemState instanceof OnOffType) {
+            bri = (short) (((OnOffType) itemState) == OnOffType.ON ? 255 : 0);
+        }
+        HueState hueState = new HueState(bri > 0, bri);
+        HueDevice d = new HueDevice(hueState, item.getLabel(), item.getName());
+        return d;
+    }
+
+    /**
+     * Gets all items that match our tag
+     *
+     * @return
+     */
+    private Collection<Item> getTaggedItems() {
+        Collection<Item> items = new LinkedList<Item>();
+        for (Item item : itemRegistry.getItems()) {
+            for (String tag : item.getTags()) {
+                if (tag.startsWith(HOMEKIT_PREFIX)) {
+                    items.add(item);
+                    break;
+                }
+            }
+        }
+        return items;
+    }
+
+    /**
+     * Checks if the username exists in our user list
+     *
+     * @param userName
+     * @return
+     * @throws IOException
+     */
+    private boolean authorizeUser(String userName) throws IOException {
+        return userNames.contains(userName);
+    }
+
+    /**
+     * Adds a username to the user file
+     *
+     * @param userName
+     * @throws IOException
+     */
+    private synchronized void addUser(String userName) throws IOException {
+        if (!userNames.contains(userName)) {
+            userNames.add(userName);
+            USER_FILE.mkdirs();
+            FileOutputStream fos = new FileOutputStream(USER_FILE);
+            IOUtils.writeLines(userNames, null, fos);
+            IOUtils.closeQuietly(fos);
```

Comment:
As per my previous comment, this has to be in a finally block in case an exception is thrown before.